.github/dependabot.yml

```diff
@@ -27,6 +27,9 @@ updates:
       emotion:
         patterns:
           - '@emotion/*'
+      esbuild:
+        patterns:
+          - '@esbuild/*'
```

Comment:
I am actually wondering whether this should be grouped with the storybook updates? As storybook seems to have a transitive dependency on it... (according to a couple of previous PRs - #12100) cc @andrewscfc

Reply:
I wonder if that makes more sense, I think we only install `esbuild` explicitly to prevent nuisance changes in the yarn cache

`yarn why esbuild`
![image](https://github.com/user-attachments/assets/fbf82a90-e77b-418b-b673-2ff25f7069c5)

Reply:
Yeah, let's try this. The only thing I am not sure about is: storybook updates can sometimes be as a result of them updating their version of esbuild. So I am not sure if there will be chicken & egg thing whereby there's a new version of the esbuild library, but because it is lumped in with storybook, it might still get out of sync...?

Let's just go with grouping with storybook initially, and see what PR dependabot generates!